### ISSUE TITLE
CSS duplication when a component is used across pages

### ISSUE THREAD

**Issue by gsm-fc** [vercel/next.js#28486]
### What version of Next.js are you using?

11.1.0

### What version of Node.js are you using?

12.19.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next start

### Describe the Bug

If I have two pages which share a common component. The CSS of the shared component is being loaded on both the pages when we switch from one page to another using client side routing (`<Link/>`). This is simple code duplication and must be fixed

### Expected Behavior

The CSS must load only once and when the user moves to the next page the CSS loaded earlier must be reused.

### To Reproduce

Clone the basic CSS example from next examples repo - https://github.com/vercel/next.js/tree/canary/examples/basic-css

Create 2 pages
1) index.js
2) about.js

Create a simple component` <User/>` whose styles are being loaded from a .module.css file

Import this component on both pages and link the pages to one another through nextjs routing.

When we open http://localhost:3000 the CSS  of index page is clubbed with CSS of user component and loaded via a CSS file as shown below
<img width="1280" alt="Screenshot 2021-08-25 at 12 57 37 PM" src="https://user-images.githubusercontent.com/80637079/130745895-cfa93900-894d-4ba0-a9d1-7c3ffb3950a8.png">
<img width="1280" alt="Screenshot 2021-08-25 at 12 57 54 PM" src="https://user-images.githubusercontent.com/80637079/130745908-a9678937-8dac-4b19-b279-f8f82427318e.png">

Note the CSS for class - "styles5_h1Custome__2VZ1Y". This is the class which is used in user component.

Now, if the user clicks on the about page link. The about page is loaded with its CSS being loaded in the head instead of a separate file.

<img width="942" alt="Screenshot 2021-08-25 at 1 01 38 PM" src="https://user-images.githubusercontent.com/80637079/130746428-080e38f9-0c17-438c-9c5d-8d788e9106b3.png">

But if we look closely in this inlined CSS we have the same CSS class "styles5_h1Custome__2VZ1Y" with same CSS for the user component.

Which leads to duplicate CSS being loaded for a given component across pages. Why was this CSS again loaded in the head when we switched the page as it is already present in the CSS file loaded with index page.

**Comment by gsm-fc** [comment]
@timneutkens @Timer Can you help?

**Comment by balazsorban44** [comment]
This issue has been automatically closed because it wasn't verified against next@canary. If you think it was closed by accident, please leave a comment. If you are running into a similar issue, please open a new issue with a reproduction. Thank you.

**Comment by maximusnikulin** [comment]
@JanKaifer Is there some progress for this problem ?

**Comment by JanKaifer** [comment]
No one provided reproduction on the canary so we just closed this.
See the comment above. Please open a new issue if you think you are experiencing a similar problem.

Please don't spam mentions. We just ignore them anyway because there are too many people tagging us.
